apps/next/app/login/page.tsx

```diff
@@ -1,10 +1,26 @@
 import { redirect } from "next/navigation"
+import * as z from "zod"
+import { zu } from "zod_utilz"
 
 import { LoginForm } from "@/components/user/login-form"
 
 import { createClient } from "@/modules/utils/server"
 
-export default async function Page() {
+const ErrorSchema = zu
```

Comment:
I'm not 100% sure if we can justify the need for the additional library just to parse the string to an object and then apply the schema.

You can create the schema object and each message and status code can have their individual catch https://zod.dev/?id=catch

We can do an if(error) parse and then render the login with the error default error message.

I know there is a bit of repetition because we are writing the component twice (with and without error) but I think it would be easier to understand just using Zod is good enough for this use case.

Reply:
Please check now. I am not sure if I understood your hints but implemented it somehow without additional deps.